airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_source.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import pydantic
+import pytest
+from airbyte_cdk.models import ConnectorSpecification
+from source_facebook_marketing import SourceFacebookMarketing
+
+
+@pytest.fixture(name="config")
+def config_fixture():
+    config = {"account_id": 123, "access_token": "TOKEN", "start_date": "2019-10-10T00:00:00"}
+
+    return config
+
+
+@pytest.fixture(name="api")
+def api_fixture(mocker):
+    api_mock = mocker.patch("source_facebook_marketing.source.API")
```

Comment:
feels a little hacky to go this deep into the code to patch it (this seems to be a common thing in python though). I wonder if dependency injection is a better pattern here.

Reply:
it is dependency injection, all imports of your module are its dependency,
It might look like a `deep into the code` but it is the module that we test (`source.py`) and its path must be absolute.